pkg/controller/controllerutil/controllerutil.go

```diff
@@ -168,6 +168,36 @@ func RemoveOwnerReference(owner, object metav1.Object, scheme *runtime.Scheme) e
 	return nil
 }
 
+// AppendOwnerReference is a helper method to make sure the given object contains a provided owner reference.
+// This allows you to declare that owner has a dependency on the object without specifying it as a controller.
+// If a reference already exists, it'll be overwritten with the newly provided version.
+func AppendOwnerReference(owner metav1.OwnerReference, object metav1.Object) {
+	upsertOwnerRef(owner, object)
+}
+
+// DropOwnerReference is a helper method to make sure the given object removes a provided owner reference.
+// This allows you to remove the owner to establish a new owner of the object in a subsequent call.
+func DropOwnerReference(owner metav1.OwnerReference, object metav1.Object) error {
+	ownerRefs := object.GetOwnerReferences()
+	index := indexOwnerRef(ownerRefs, metav1.OwnerReference{
+		APIVersion: owner.APIVersion,
+		Name:       owner.Name,
+		Kind:       owner.Kind,
+	})
```

Comment:
We usually do this with GVK because relying on this may not be stable.  (The workaround exist because of issues around this).  Getting the GVK requires the scheme. Adding the scheme makes this method the `RemoveOwnerReference` which already exists. 

I am trying to understand the repeat logic around this functionality.  What are we trying to circumvent utilizing this logic?

Reply:
Could you elaborate on that instability? Is that documented somewhere?

Reply:
Discussed here: https://github.com/kubernetes-sigs/controller-runtime/pull/2882#discussion_r1681547839
Reference:
https://github.com/kubernetes/kubernetes/issues/3030
https://github.com/kubernetes/kubernetes/issues/80609

I am still trying to understand what you are trying to do because this looks very similar to what is already in `controllerutil` and conflating the two different functions where the logic is the same can cause confusion on what we are trying to achieve.

Reply:
This function I added just so there was a pair of functions that work with OwnerReference instead of Object. I don't personally need the DropOwnerReference function.